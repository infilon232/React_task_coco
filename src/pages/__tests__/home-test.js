import React from "react";
import { FlatList, Text } from "react-native";
import renderer from "react-test-renderer";
import Home from "../index";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { act, waitFor } from "@testing-library/react-native";

jest.mock("@react-navigation/native", () => {
  return {
    useFocusEffect: (callback) => callback(),
    useIsFocused: () => true
  };
});

const initialState = {
  categories: { name: "sports", id: "1" },
  id: "1",
  name: "sam",
  profile: "https://picsum.photos/id/1000/224/224"
};

const mockStore = configureStore();
const store = mockStore(initialState);
describe("Home", () => {
  const fakeNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn()
  };
  jest.useFakeTimers();

  it("should render correctly", () => {
    let rendered;
    act(() => {
      rendered = renderer
        .create(
          <Provider store={store}>
            <Home navigation={fakeNavigation} />
          </Provider>
        )
        .toJSON();
    });
    waitFor(() => {
      expect(rendered).toMatchSnapshot();
    });
  });

  jest.mock("@react-native-firebase/database", () => {
    const set = jest.fn();
    return {
      database: jest.fn(() => ({
        ref: jest.fn(() => ({
          once: jest.fn(() => ({
            then: jest.fn(() => ({
              set
            }))
          }))
        }))
      }))
    };
  });

  it("renders the FlatList component", () => {
    const tree = renderer
      .create(
        <FlatList
          data={["apple", "banana", "kiwi"]}
          keyExtractor={(item) => item}
          renderItem={({ item }) => <Text>{item}</Text>}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
