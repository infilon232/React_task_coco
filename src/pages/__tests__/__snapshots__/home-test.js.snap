// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the FlatList component 1`] = `
<RCTScrollView
  data={
    Array [
      "apple",
      "banana",
      "kiwi",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        apple
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        banana
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        kiwi
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Home should render correctly 1`] = `undefined`;

exports[`renders correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#000",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        ListHeaderComponent={[Function]}
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        keyboardShouldPersistTaps="handled"
        nestedScrollEnabled={true}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
          >
            <View
              style={Object {}}
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`renders the FlatList component 1`] = `
<RCTScrollView
  data={
    Array [
      "apple",
      "banana",
      "kiwi",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        apple
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        banana
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        kiwi
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
